django/urls/resolvers.py

```diff
@@ -478,11 +478,24 @@ def _reverse_with_prefix(self, lookup_view, _prefix, *args, **kwargs):
             lookup_view_s = lookup_view
 
         patterns = [pattern for (possibility, pattern, defaults) in possibilities]
-        raise NoReverseMatch(
-            "Reverse for '%s' with arguments '%s' and keyword "
-            "arguments '%s' not found. %d pattern(s) tried: %s" %
-            (lookup_view_s, args, kwargs, len(patterns), patterns)
-        )
+
+        if patterns:
+            if args:
+                arg_msg = "arguments '%s'" % (args,)
+            elif kwargs:
+                arg_msg = "keyword arguments '%s'" % (kwargs,)
+            else:
+                arg_msg = "no arguments"
+            msg = (
+                "Reverse for '%s' with %s not found. %d pattern(s) tried: %s" %
+                (lookup_view_s, arg_msg, len(patterns), patterns)
+            )
+        else:
+            msg = (
+                "Reverse for '%(view)s' not found. '%(view)s' is not "
+                "a valid view function or pattern name" % {'view': lookup_view_s}
```

Comment:
I'm making of few cosmetic edits. I wonder if it would be more clear to say "url() name" rather than "pattern name" here?